File: src/main/java/PhoneTrack.java
import spark.Request;
import spark.Response;

import java.sql.SQLException;
import java.sql.Statement;
public class PhoneTrack {
    public PhoneTrack() {

    }

    public String addData(Request request, Response response) {
        try {
            String tst = Mqtt.getMysqlDateString(Long.parseLong(request.queryParams("timestamp")));
            Statement stmt = Mysql.conn.createStatement();
            String alt = request.queryParams("alt");
            if (alt.equals("")){
                stmt.executeUpdate("INSERT INTO data VALUES (NULL, '" + tst + "', '"
                        + request.queryParams("lat") + "', '" + request.queryParams("lon") + "'" +
                        ", NULL, '" + request.queryParams("acc")
                        + "',  '" + request.queryParams("batt") + "', '" + request.queryParams("tid") + "')");
            } else {
                stmt.executeUpdate("INSERT INTO data VALUES (NULL, '" + tst + "', '"
                        + request.queryParams("lat") + "', '" + request.queryParams("lon") + "'" +
                        ", '" + alt + "', '" + request.queryParams("acc")
                        + "',  '" + request.queryParams("batt") + "', '" + request.queryParams("tid") + "')");
            }

            stmt.close();
        } catch (Exception e) {
            e.printStackTrace();
            try {
                System.out.println(Mysql.conn.isValid(3000));
            } catch (SQLException ex) {
                ex.printStackTrace();
            }
            response.status(500);
        }
        return "";
    }
}

File: src/main/java/PhoneTrack.kt
import io.ktor.application.*
import io.ktor.http.*
import io.ktor.response.*
import io.ktor.routing.*
import java.lang.Exception
import java.sql.SQLException

fun Route.addPhoneTrackLocation(mysql: Mysql){
    route("/add") {
        get {
            try {
                val tst = Mqtt.getMysqlDateString(call.parameters["timestamp"]!!.toLong())
                val stmt = Mysql.conn.createStatement()
                val alt = call.parameters["alt"]!!
                if (alt == "") {
                    stmt.executeUpdate(
                        "INSERT INTO data VALUES (NULL, '" + tst + "', '"
                                + call.parameters["lat"]!! + "', '" + call.parameters["lon"]!! + "'" +
                                ", NULL, '" + call.parameters["acc"]!!
                                + "',  '" + call.parameters["batt"]!! + "', '" + call.parameters["tid"]!! + "')"
                    )
                } else {
                    stmt.executeUpdate(
                        ("INSERT INTO data VALUES (NULL, '" + tst + "', '"
                                + call.parameters["lat"]!! + "', '" + call.parameters["lon"]!! + "'" +
                                ", '" + alt + "', '" + call.parameters["acc"]!!
                                + "',  '" + call.parameters["batt"]!! + "', '" + call.parameters["tid"]!! + "')")
                    )
                }
                stmt.close()
            } catch (e: Exception) {
                e.printStackTrace()
                try {
                    println(Mysql.conn.isValid(3000))
                } catch (ex: SQLException) {
                    ex.printStackTrace()
                }
                call.respondText("Insert failed", status = HttpStatusCode.InternalServerError)
            }
        }
    }
}